sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -112,5 +125,134 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
+                        .Append(": ")
+                        .AppendLine(info.Value);
+                }
+            }
+
+            if (response.ContentStream is MemoryStream)
+            {
+                string content = response.Content.ToString();
+                if (content.Length > 0)
+                {
+                    messageBuilder
+                        .AppendLine()
+                        .AppendLine("Content:")
+                        .AppendLine(content);
+                }
+            }
+
+            messageBuilder
+                .AppendLine()
+                .AppendLine("Headers:");
+
+            foreach (HttpHeader responseHeader in response.Headers)
+            {
+                string headerValue = response.Sanitizer.SanitizeHeader(responseHeader.Name, responseHeader.Value);
+                messageBuilder.AppendLine($"{responseHeader.Name}: {headerValue}");
+            }
+
+            var formatMessage = messageBuilder.ToString();
+            return (formatMessage, error?.Code, data);
+        }
+
+        /// <summary>
+        /// This is intentionally sync-only as it will only be called by the ctor.
+        /// </summary>
+        /// <param name="response"></param>
+        /// <returns></returns>
+        internal static string? ReadContent(Response response)
+        {
+            string? content = null;
+
+            if (response.ContentStream != null &&
+                ContentTypeUtilities.TryGetTextEncoding(response.Headers.ContentType, out var encoding))
+            {
+                using (var streamReader = new StreamReader(response.ContentStream, encoding))
+                {
+                    content = streamReader.ReadToEnd();
+                }
+            }
+
+            return content;
+        }
+
+        internal static bool TryExtractErrorContent(Response response, out ResponseError? error, out IDictionary<string, string>? data)
+        {
+            error = null;
+            data = null;
+            try
+            {
+                string? content = null;
+                if (response.ContentStream != null && response.ContentStream.CanSeek)
+                {
+                    content = response.Content.ToString();
+                }
+                else
+                {
+                    // this path should only happen in exceptional cases such as when
+                    // the RFE ctor was called directly by client or customer code with an un-buffered response.
+                    // Generated code would never do this.
+                    content = ReadContent(response);
+                }
+                // Optimistic check for JSON object we expect
+                if (content == null || !content.StartsWith("{", StringComparison.OrdinalIgnoreCase))
+                {
+                    return false;
+                }
+                error = System.Text.Json.JsonSerializer.Deserialize<ErrorResponse>(content)?.Error;
```

Comment:
nit: avoid fully qualifying

Reply:
This was to avoid conflicts with the `System.Runtime.Serialization`